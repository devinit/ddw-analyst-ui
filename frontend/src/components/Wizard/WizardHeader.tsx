import React, { FC } from 'react';
import { Card } from 'react-bootstrap';

type ComponentProps = {
  children?: React.ReactNode;
};
const WizardHeader: FC<ComponentProps> = ({ children }) => {
  return <Card.Header className="text-center">{children}</Card.Header>;
};

export { WizardHeader };
