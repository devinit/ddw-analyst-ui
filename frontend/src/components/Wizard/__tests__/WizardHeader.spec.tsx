/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { WizardHeader } from '../WizardHeader';

describe('WizardHeader', () => {
  test('renders the default structure correctly', () => {
    const renderer = TestRenderer.create(<WizardHeader />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <WizardHeader>
        <div>My Child</div>
      </WizardHeader>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
