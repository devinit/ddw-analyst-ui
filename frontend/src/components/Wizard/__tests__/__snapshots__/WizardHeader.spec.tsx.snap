// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardHeader renders the default structure correctly 1`] = `
<div
  className="text-center card-header"
/>
`;

exports[`WizardHeader renders with children 1`] = `
<div
  className="text-center card-header"
>
  <div>
    My Child
  </div>
</div>
`;
